### ISSUE TITLE
Keras Custom Functions

### ISSUE THREAD

**Issue by MadihaAK** [keras-team/keras#19044]
I have recently started building Neural Networks and I am currently working on a project to develop a NN for Van der Pol's equation. My problem is that I developed a sequential model with time and mu as inputs. But my loss function is custom (because I needed to compute derivatives etc) and it requires additional external custom parameters (2 arrays of input data and one scalar), which change based on input data. My question is that, I have trained my model, but before predicting, I have to update these loss function parameters (external ones), so do I use model.compile before predicting? Because I have read alot and from what I assume, it degrades the performance of the trained model. Is there any other way to achieve this? please help.

**Comment by MadihaAK** [comment]
@SuryanarayanaY yes exactly, I have already trained my model and now I want to predict an entirely new dataset. I tried what you said about reloading, and it helps (lesser losses for training data) however, my issue is with the loss function. You said model.predict does not call loss function so that means my stored values (or weights) derived based on training data will be used for the other data I test. That is where my concern is, because the weights won't be updated again, but as my loss function is dependent upon inputs and outputs, logically it should update the model when I give a new dataset. I don't know if its doable though, I will try to predict regardless and see the difference

**Comment by SuryanarayanaY** [comment]
Hi @MadihaAK ,

`model.predict` will take inputs and give outputs directly but never update weights.To make the model more robust we use validation_data to test and train the model until we get desired result of validation metrics.

You may customize `train_step` and loss_function to configure the custom parameters and train the model again. You can refer this [guide](https://keras.io/guides/writing_a_custom_training_loop_in_tensorflow/) for custom training demo.

Thanks!
